File: iPhone_TNG/NextGenRepliconTimeSheet/DataObjects/Violation.m
#import "Violation.h"


@interface Violation ()

@property (nonatomic) NSString *title;
@property (nonatomic) ViolationSeverity severity;
@property (nonatomic) Waiver *waiver;

@end


@implementation Violation

- (instancetype)initWithSeverity:(ViolationSeverity)severity
                          waiver:(Waiver *)waiver
                           title:(NSString *)title;
{
    self = [super init];
    if (self) {
        self.title = title;
        self.severity = severity;
        self.waiver = waiver;
    }
    return self;
}

@end

File: iPhone_TNG/NextGenRepliconTimeSheetUITests/Data Models/Violation.swift

import Foundation

class Violation{

    var title   : String = ""
    var waiver  : Waiver?
    var type    : String = ""
    var defaultStatus : String = "No Response"

    init(title : String , waiver : Waiver, type: String) {
        self.waiver = waiver
        self.title = title
        self.type = type

    }
    
}